### ISSUE TITLE
Tests for range proof on multi-party commitment

### ISSUE THREAD

**Issue by garyyu** [mimblewimble/rust-secp256k1-zkp#31]
After adding this feature, we don't have any tests.

```
    pub fn bullet_proof(
        &self,
        value: u64,
        blind: SecretKey,
        nonce: SecretKey,
        extra_data: Option<Vec<u8>>,
        message: Option<ProofMessage>,
    ) -> RangeProof {
        ...
        // TODO: expose multi-party support
        let tau_x = ptr::null_mut();
        let t_one = ptr::null_mut();
        let t_two = ptr::null_mut();
        let commits = ptr::null_mut();
        let private_nonce = ptr::null();

```

@jaspervdm Could you please give me some examples to use this new feature?  so as me to add some tests for it.

**Comment by jaspervdm** [comment]
The library includes tests for the multi-party proof, see [here](https://github.com/mimblewimble/secp256k1-zkp/blob/master/src/modules/bulletproofs/tests_impl.h#L569)

**Comment by garyyu** [comment]
@jaspervdm  more test cases has been added:  https://github.com/mimblewimble/rust-secp256k1-zkp/pull/37

And one problem found: `rewind_bullet_proof()` will fail on this new range proof. Any idea on rewinding?
```
		// 5. Rewind message with same blinding factor
		let commit = secp
			.commit_sum(vec![partial_commit_a, partial_commit_b], vec![])
			.unwrap();
		let proof_info = secp.rewind_bullet_proof(
			commit,
			common_nonce.clone(),
			Some(extra_data.to_vec()),
			bullet_proof,
		);
		println!("proof_info after rewind: {:#?}", proof_info);
```

**Comment by garyyu** [comment]
`rewind_bullet_proof()` will not support at the moment. Refer to detail info in: https://github.com/mimblewimble/grin/issues/1601

**Comment by garyyu** [comment]
Closed by #37
